india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py

```diff
@@ -0,0 +1,368 @@
+                journal_entry.name.as_("voucher_no"),
+                journal_entry.posting_date,
+                Sum(
+                    Case()
+                    .when(
+                        journal_entry_account.account == self.gst_accounts.igst_account,
+                        (-1 * journal_entry_account.credit_in_account_currency),
+                    )
+                    .else_(0)
+                ).as_("integrated_tax"),
+                Sum(
+                    Case()
+                    .when(
+                        journal_entry_account.account == self.gst_accounts.cgst_account,
+                        (-1 * journal_entry_account.credit_in_account_currency),
+                    )
+                    .else_(0)
+                ).as_("central_tax"),
+                Sum(
+                    Case()
+                    .when(
+                        journal_entry_account.account == self.gst_accounts.sgst_account,
+                        (-1 * journal_entry_account.credit_in_account_currency),
+                    )
+                    .else_(0)
+                ).as_("state_tax"),
+                Sum(
+                    Case()
+                    .when(
+                        journal_entry_account.account == self.gst_accounts.cess_account,
+                        (-1 * journal_entry_account.credit_in_account_currency),
+                    )
+                    .else_(0)
+                ).as_("cess_amount"),
+                journal_entry.reversal_type.as_("eligibility_for_itc"),
+            )
+            .where(
+                (journal_entry.docstatus == 1)
+                & (journal_entry.posting_date[self.from_date : self.to_date])
+                & (journal_entry.company == self.company)
+                & (journal_entry.company_gstin == self.company_gstin)
+                & (journal_entry.voucher_type == "Reversal of ITC")
+            )
+            .groupby(journal_entry.name)
+        )
+        return query.run(as_dict=True)
+
+
+class GSTR3B_Inward_Nil_Exempt(BaseGSTR3BDetails):
+    def extend_columns(self):
+        self.columns.extend(
+            [
+                {
+                    "fieldname": "intra",
+                    "label": _("Intra State"),
+                    "fieldtype": "Currency",
+                    "width": 100,
+                },
+                {
+                    "fieldname": "inter",
+                    "label": _("Inter State"),
+                    "fieldtype": "Currency",
+                    "width": 100,
+                },
+                {
+                    "fieldname": "nature_of_supply",
+                    "label": _("Nature of Supply"),
+                    "fieldtype": "Data",
+                    "width": 100,
+                },
+            ]
+        )
+
+    def get_data(self):
+        formatted_data = []
+
+        invoices = self.get_inward_nil_exempt()
+
+        address_state_map = self.get_address_state_map()
```

Comment:
from document

Reply:
state is saved in display field , address template can change.